x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorPlugin.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ilm;
+
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.env.NodeEnvironment;
+import org.elasticsearch.health.HealthIndicatorService;
+import org.elasticsearch.plugins.HealthIndicatorPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.repositories.RepositoriesService;
+import org.elasticsearch.script.ScriptService;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.watcher.ResourceWatcherService;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class IlmHealthIndicatorPlugin extends Plugin implements HealthIndicatorPlugin {
+
+    private final SetOnce<IlmHealthIndicatorService> healthIndicatorService = new SetOnce<>();
+
+    @Override
+    public Collection<Object> createComponents(
+        Client client,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ResourceWatcherService resourceWatcherService,
+        ScriptService scriptService,
+        NamedXContentRegistry xContentRegistry,
+        Environment environment,
+        NodeEnvironment nodeEnvironment,
+        NamedWriteableRegistry namedWriteableRegistry,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<RepositoriesService> repositoriesServiceSupplier
+    ) {
+        var service = new IlmHealthIndicatorService(clusterService);
+        healthIndicatorService.set(service);
+        return List.of(service);
```

Comment:
I would prefer to just keep a reference to the `ClusterService` and then create the health-indicator-service in the method below instead. There is no reason to create it early and in particular no reason AFAICS to expose the component as an inject-able component here.

Reply:
I guess we could not expose it here (and return an empty list), however I would still prefer to init it here rather then create a service object every time we make a rest call

Reply:
I think we should only call this once at bootstrap and AFAICS, that is also the case with the current code, i.e., it is called once and then the HealthService is instantiated here:

https://github.com/elastic/elasticsearch/pull/83205/files#diff-d01b960cab84fe2338f230b5e7535bc8efa9f9778ecf61c9f527067351dca895R910